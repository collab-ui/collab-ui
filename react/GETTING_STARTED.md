## Getting Started

It's fairly simple and easy. Just follow three simple steps as explained below:

### 1. Fork the repo

### 2. Clone the repo

Clone the `momentum-ui-react` git repo:

```sh
git clone git@github.com:momentum-ui/momentum-ui-react.git
```

### 3. Install all dependencies & devdependencies

From the command line navigate to 'momentum-ui-react' directory created in previous step
and run following command:

```sh
yarn
```

### 4. Start Docs Site

From the command line run the code
as below:

```sh
yarn start
```

Behind the scene above command does all the heavy lifting which can be summarized as below:

* Building ES6/JSX source and output to `dist` folder
* Watching `src` folder to build on every changes
* Bundling docs application using webpack

There are npm scripts available for each of these individual task that alone can be run. Refer `package.json` for more details.

Now you can try editing source in src folder. With every changes in source saved, the demo app automagically updates without hitting F5 or browser reload button.
This setup provide a playground environment and also act as a demo of components showing their typical usage with various source examples.

### 5.  Build

To build the project for production, use following npm script:

```sh
yarn build
```

If everything goes fine, folders [`dist`, `bundles`, `lib`] will be created in root folder of this project. Folder `lib` will have minified ES6 modules with similar directory structure as source.
Generated minified ES6 files can be directly imported in ES6 application source code.

## Codebase Directory Structure

The React Toolkit codebase has following folder structure:

```sh
momentum-ui-react
│  CHANGELOG.md
│  CONTRIBUTING.md
│  GETTING_STARTED.md
│  README.md
│  LICENSE
│  package.json
|  .babelrc
|  .editorconfig
|  .eslintignore
|  .eslintrc
|  .gitignore
│  .istanbul.yml
│  .npmignore
|  .npmrc
│  .watchmanconfig
|  ...
└───src
│  │
│  └───lib
│  |  └───Button
|  |  |  |  index.js
|  |  |  |  index.spec.js
|  |  |  |  __snapshots__
|  │  |  |  └───index.spec.js.snap
|  |  |  |  ...
│  |  └───Select
|  |  |  |  ...
|  |  |  index.js
│  └───docs
|  |  |  ...
│  └───helpers
└───tools
|  │  script.js
|  |  ...
└───config
|  |  webpack.config.dev.js
|  |  ...
```

### About directory structure

* All source goes inside `src` folder written using ES6/JSX standard.

* For every component there is a folder inside `src/lib`.

* Each component folder (say button) houses JSX source code and `__snapshots__` folder.

At the time of publishing only minimal files and folders are included in module to keep download / installation faster and save space on your disk. Post installation of colab-ui-react NPM
module it will have following directory structure:

```sh
momentum-ui-react
│  README.md
│  LICENSE
│  package.json
|  index.js
|  ...
└───lib
|  └───Button
|  |  |  index.js
|  |  |  ...
|  └───Select
|  |  |  ...
|  |  ...
|
└───bundles
|  |  ...
└───dist
|  |  ...
```

As evident above, only assets (dist, bundles, lib) are part of published module.

## References & Links

[Contribution guidelines](/CONTRIBUTING.md)
