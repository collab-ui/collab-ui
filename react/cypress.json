{
  "testFiles": "**/*.cy.js",
  "integrationFolder": "src/lib",
  "numTestsKeptInMemory": 1,
  "pluginsFile": "test/plugins/index.js",
  "supportFile": "test/support/index.js",
  "fixturesFolder": "test/fixtures",
  "chromeWebSecurity": false,
  "env": {
    "BASE_URL": "http://127.0.0.1:4300"
  },
  "video": false
}
