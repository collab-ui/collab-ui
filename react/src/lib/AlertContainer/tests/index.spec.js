import React from 'react';
import { shallow, mount } from 'enzyme';
import { Alert, AlertContainer } from '@collab-ui/react';

describe('tests for <AlertContainer />', () => {
  const alertTitle = 'Now Hear This!';
  const alertMessage = 'Jefe Guadelupe, unit tesing like a boss!';

  it('should match SnapShot', () => {
    const container = shallow(<AlertContainer/>);

    expect(container).toMatchSnapshot();
  });

  it('should render a div in bottom-right by default', () => {
    const container = shallow(<AlertContainer />);

    expect(container.find('.cui-alert__container--bottom-right').length).toEqual(1);
  });

  it('should honor position prop when top-left is passed in', () => {
    const container = shallow(<AlertContainer position={'top-left'}/>);

    expect(container.find('.cui-alert__container--top-left').length).toEqual(1);
  });

  it('should honor position prop when top-center is passed in', () => {
    const container = shallow(<AlertContainer position={'top-center'}/>);

    expect(container.find('.cui-alert__container--top-center').length).toEqual(1);
  });

  it('should honor position prop when top-right is passed in', () => {
    const container = shallow(<AlertContainer position={'top-right'}/>);

    expect(container.find('.cui-alert__container--top-right').length).toEqual(1);
  });

  it('should honor position prop when bottom-left is passed in', () => {
    const container = shallow(<AlertContainer position={'bottom-left'}/>);

    expect(container.find('.cui-alert__container--bottom-left').length).toEqual(1);
  });

  it('should honor position prop when bottom-center is passed in', () => {
    const container = shallow(<AlertContainer position={'bottom-center'}/>);

    expect(container.find('.cui-alert__container--bottom-center').length).toEqual(1);
  });

  it('should honor position prop when bottom-right is passed in', () => {
    const container = shallow(<AlertContainer position={'bottom-right'}/>);

    expect(container.find('.cui-alert__container--bottom-right').length).toEqual(1);
  });

  it('should render an info Alert when info() is called', () => {
    const container = mount(<AlertContainer/>);
    container.instance().info(
      alertTitle,
      alertMessage,
      () => {},
      { ariaLabel: 'Close' }
    );
    container.update();
    expect(container.find('.cui-alert--info').length).toEqual(1);
  });

  it('should render a success Alert when success() is called', () => {
    const container = mount(<AlertContainer/>);
    container.instance().success(
      alertTitle,
      alertMessage,
      () => {},
      { ariaLabel: 'Close' }
    );
    container.update();
    expect(container.find('.cui-alert--success').length).toEqual(1);
  });

  it('should render a warning Alert when warning() is called', () => {
    const container = mount(<AlertContainer/>);
    container.instance().warning(
      alertTitle,
      alertMessage,
      () => {},
      { ariaLabel: 'Close' }
    );
    container.update();
    expect(container.find('.cui-alert--warning').length).toEqual(1);
  });

  it('should render an error Alert when error() is called', () => {
    const container = mount(<AlertContainer/>);
    container.instance().error(
      alertTitle,
      alertMessage,
      () => {},
      { ariaLabel: 'Close' }
    );
    container.update();
    expect(container.find('.cui-alert--error').length).toEqual(1);
  });

  it('number of Alerts should equal number of times functions are called', () => {
    const container = mount(<AlertContainer/>);
    const alertCount = 5;

    for (let index = 0; index < alertCount; index++) {
      container.instance().info(
        alertTitle,
        alertMessage,
        () => {},
        { ariaLabel: 'Close' }
      );
    }
    container.update();
    expect(container.find(Alert).length).toEqual(alertCount);
  });


  it('should pass any other HTML props to Alert', () => {
    const container = mount(<AlertContainer/>);
    container.instance().info(
      alertTitle,
      alertMessage,
      () => {},
      { ariaLabel: 'Close', id: 'testProp' }
    );
    container.update();
    expect(container.find('button').props().id).toEqual('testProp');
  });
});
