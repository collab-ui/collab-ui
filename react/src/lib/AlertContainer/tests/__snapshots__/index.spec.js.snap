// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <AlertContainer /> should match SnapShot 1`] = `
<div
  className="md-alert__container md-alert__container--bottom-right"
  role="alert"
/>
`;
